src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift

```diff
@@ -687,12 +687,10 @@ class HealthCertificateService: HealthCertificateServiceServable {
 		}
 	}
 
-	private func updateValidityState(for healthCertificate: HealthCertificate) {
+	private func updateValidityState(for healthCertificate: HealthCertificate, person: HealthCertifiedPerson) {
 		let previousValidityState = healthCertificate.validityState
 
-		let blockedIdentifierChunks = appConfiguration.currentAppConfig.value
-			.dgcParameters.blockListParameters.blockedUvciChunks
-		if healthCertificate.isBlocked(by: blockedIdentifierChunks) {
+		if let invalidationRules = person.dccWalletInfo?.certificatesRevokedByInvalidationRules, healthCertificate.isBlocked(by: invalidationRules) {
```

Comment:
If I would keep it in a separate function, I would probably move it to the person now, but I don't think it makes much sense to have this separate function anymore as it is just a `contains` call.

```suggestion
		if let revokedCertificates = person.dccWalletInfo?.certificatesRevokedByInvalidationRules, 
			revokedCertificates.contains(where: { $0.certificateRef.barcodeData == healthCertificate.base45 }) {
```

Unit tests (which still seem to be to do for this PR) will then have to be done with the service unit tests.